### DISCUSSION TITLE
Code coverage for Angular 17 (esbuilder) and E2E Cypress tests

### DISCUSSION THREAD

**Discussion by AlexKorole** [cypress-io/cypress#28463] [Questions and Help]
Hello, everyone. Angular 17 advice a new builder - application or browser-esbuild ([https://angular.io/guide/esbuild](https://angular.io/guide/esbuild)) , but with this builder doesn't work feature for meagering code coverage of Angular project by Cypress E2E tests. We tried to use follow: [https://github.com/skylock/cypress-angular-coverage-example/blob/master/README.md](ngx-build-plus) , but it doesn't work with new builder. Does cypress have some other ways to measure coverage for Angular 17(esbuild) project?

**Comment by mpontreau** [comment]
Did you find any way to set up the code coverage with the angular builder ?

**Comment by AlexKorole** [comment]
Now for our project works only this way: [https://github.com/skylock/cypress-angular-coverage-example/blob/master/README.md](https://github.com/skylock/cypress-angular-coverage-example/blob/master/README.md)
Also I tried to do in such way [https://lukas-klement.medium.com/implementing-code-coverage-with-angular-and-cypress-6ed08ed7e617](https://lukas-klement.medium.com/implementing-code-coverage-with-angular-and-cypress-6ed08ed7e617) , but it  doesn't work for our project.

**Comment by mpontreau** [comment]
Well our project was already based on the second link you mentionned.
For the first one I give it a try but it seems to also use webPack and the esbuild still not working.
You really have the new angular esbuild with a coverage ?

**Reply by AlexKorole** [reply]
How to do it in Angular 17(esbuild) project was my question at the beginning of this request. Currently we use Angular 17(webpack) because of problem with measuring coverage by  Cypress E2E tests.

**Comment by mschoneman** [comment]
It's an awful hack, but I added a "build-coverage" section that was just used for code coverage and used the `@angular-builders/custom-webpack:browser` while still using the `@angular-devkit/build-angular:application` to actually build the release of the app.

**Reply by bauerbua** [reply]
This is what worked for me aswell. I am just wondering if there are any potential risks in doing it that way? Like could some tests, that pass with `@angular-builders/custom-webpack:browser` actually fail in production with `@angular-devkit/build-angular:application`?

**Reply by mpontreau** [reply]
yes it can, but hopefully it should not.
For example if you have an early version of angular 17 you can face some issues with routing on vite:
https://github.com/angular/angular-cli/issues/26618#issuecomment-1880236486

That's why it would be nice to run code coverage on esbuild server.

**Comment by DanyilBondarenkoNvable** [comment]
Hello! Sorry, any updated on the topic? We have recently updated to the new builder and we are still struggling with the setup of test coverage.
